go/apps/linux_host/linux_host.go

```diff
@@ -17,247 +17,534 @@ package main
 
-		switch *action {
-		case "show":
-			fmt.Printf("%v", host.HostName())
-		case "start":
-			// Make sure callers can read the directory that
-			// contains the socket.
-			err := os.Chmod(path.Dir(sockPath), 0755)
-			fatalIf(err)
-
-			// The Serve method on the linux host admin server
-			// requires a UnixListener, since it uses this listener
-			// to get the UID and GID of callers. So, we have to use
-			// the Unix-based net functions rather than the generic
-			// ones.
-			uaddr, err := net.ResolveUnixAddr("unix", sockPath)
-			fatalIf(err)
-			sock, err := net.ListenUnix("unix", uaddr)
-			fatalIf(err)
-			defer sock.Close()
-			err = os.Chmod(sockPath, 0666)
-			fatalIf(err)
-
-			fmt.Fprintf(verbose, "Linux Tao Service (%s) started and waiting for requests\n", host.HostName())
-			tao.NewLinuxHostAdminServer(host).Serve(sock)
+		vmMemory := cfg.GetKvmCoreosVmMemory()
+		if vmMemory == 0 {
+			vmMemory = 1024
 		}
-	case "shutdown":
-		glog.Fatal("not yet implemented")
-	default:
+
+		cfg := &tao.CoreOSConfig{
+			ImageFile:  coreOSImage,
+			Memory:     int(vmMemory),
+			RulesPath:  rulesPath,
+			SSHKeysCfg: sshKeysCfg,
+		}
+		childFactory, err = tao.NewLinuxKVMCoreOSFactory(socketPath, cfg)
+		failIf(err, "Can't create KVM CoreOS factory")
+	}
+
+	var host *tao.LinuxHost
+	var err error
+	switch tc.HostType {
+	case tao.Root:
+		pwd := getKey("root host key password", "pass")
+		host, err = tao.NewRootLinuxHost(hostPath(), domain.Guard, pwd, childFactory)
+	case tao.Stacked:
+		parent := tao.ParentFromConfig(tc)
+		if parent == nil {
+			usage("No host tao available, verify -parent_type or $%s\n", tao.HostChannelTypeEnvVar)
+		}
+		host, err = tao.NewStackedLinuxHost(hostPath(), domain.Guard, tao.ParentFromConfig(tc), childFactory)
+	}
+	failIf(err, "Can't create host")
+
+	return host
+}
+
+func initHost(domain *tao.Domain) {
+	var cfg tao.LinuxHostConfig
+
+	configureFromOptions(&cfg)
+	_ = loadHost(domain, &cfg)
+
+	// If we get here, keys were created and flags must be ok.
+
+	file, err := util.CreatePath(hostConfigPath(), 0777, 0666)
+	failIf(err, "Can't create host configuration")
+	cs := proto.MarshalTextString(&cfg)
+	fmt.Fprint(file, cs)
+	file.Close()
+}
+
+func showHost(domain *tao.Domain) {
+	cfg := configureFromFile()
+	configureFromOptions(cfg)
+	host := loadHost(domain, cfg)
+	fmt.Printf("%v\n", host.HostName())
+}
+
+func isBoolFlagSet(name string) bool {
+	f := flag.Lookup("logtostderr")
+	if f == nil {
+		return false
+	}
+	v, ok := f.Value.(flag.Getter).Get().(bool)
+	return ok && v
+}
+
+func daemonize() {
+	// For our purposes, "daemon" means being a session leader.
+	sid, _, errno := syscall.Syscall(syscall.SYS_GETSID, 0, 0, 0)
+	var err error
+	if errno != 0 {
+		err = errno
+	}
+	failIf(err, "Can't get process SID")
+	if int(sid) != syscall.Getpid() {
+		if syscall.Getpid() == syscall.Getpid() {
+			fmt.Fprintf(noise, "Forking to enable setsid\n")
+		} else {
+			fmt.Fprintf(noise, "Forking anyway\n")
```

Comment:
Please make this output a little more detailed, since I don't think I would know why it was "anyway"